containerd-shim-spin/src/engine.rs

```diff
@@ -140,6 +136,30 @@ impl SpinEngine {
                                 .write_data(&artifact.layer, &artifact.config.digest())
                                 .await?;
                         }
+                        MediaType::Other(name) if name == spin_oci::client::ARCHIVE_MEDIATYPE => {
+                            log::debug!(
+                                "<<< writing archive layer and unpacking contents to cache, near {:?}",
+                                cache.manifests_dir()
+                            );
+
+                            // cache.unpack_archive_layer attempts to create a tempdir via tempfile::tempdir()
+                            // which will fall back to /tmp if TMPDIR is not set. /tmp is either not found or not accessible
+                            // in the shim environment, hence setting to current working directory.
+                            if env::var("TMPDIR").is_err() {
+                                log::debug!(
+                                    "<<< TMPDIR is not set; setting to current working directory for unpacking archive layer"
+                                );
+                                env::set_var("TMPDIR", ".");
```

Comment:
I believe this is because we are using a scratch empty dir for configuring the containers root dir that the shim is running in so there isn't anything in it at all.

This generally makes sense as a solution.  Is there any concern with using `.` which is relative?  I don't think there is but might be better to be protective and use `env::current_dir()`?

Reply:
Thanks @jsturtevant, that definitely seems better.  I used an `unwrap_or(".")` but maybe we should just hard error if env::current_dir() errors?

Reply:
`unwrap_or(".")` works for me 👍